### ISSUE TITLE
[UX] Sort views alphabetically

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#521]
![structure-views_menu_makes_no_sense](https://cloud.githubusercontent.com/assets/2423362/5591356/f4ac78aa-91aa-11e4-8142-aa0104e86099.png)

1. The views submenu entries in the menu are listed in alphabetical order while in the views list page they are not.
2. They are not grouped together nor listed alphabetically with the rest of the menu entries:
  - The "Add new view", "Import", "List" and "Settings" entries are listed alphabetically, but not when views are added (once enabled)
  - The views are listed alphabetically, but the "Settings" entry is included/sorted along and the alphabetization starts after the "Add new view", "Import", "List" trio.

WTF!?!

Did we mean to list the views within the "List" entry but failed somehow? If not, I think it would make sense to move them there.

If we do not move them under "List", then lets move all non-view-name entries in that menu (currently "Add new view", "Import", "List" and "Settings") at the top and list the views entries alphabetically right after them.

I think that if the views entries are to be listed alphabetically in the menu, then so should they be in the index page (```/admin/structure/views```). If we choose to not list them alphabetically in the views index page, then at least keep the same order in the menu. If we do decide to list them alphabetically everywhere (my vote BTW), then the enabled views in the views index page should "bubble" up and the disabled kept down, but still in alphabetical order.

**Comment by quicksketch** [comment]
> Did we mean to list the views within the "List" entry but failed somehow? If not, I think it would make sense to move them there.

This is a straight-port from D7's admin_menu. I don't think we've reconsidered anything about the way that views are listed in the menu.

My preference would be to simply remove the individual Views from the admin bar entirely. We don't include blocks, layouts, text formats, image styles, or any other user-created configurations within a sub-system directly within the admin bar, why are Views special? You can easily end up with dozens of views even on a moderately complex site, making use of the admin menu to navigate them hopeless anyway.

**Comment by klonos** [comment]
> ...why are Views special?

I might have had something to do with it: https://www.drupal.org/node/884032

**Comment by klonos** [comment]
...I've found a whole lot wrong with the order of the entries in the admin menu and crating a meta-issue BTW. Give me some time.

**Comment by quicksketch** [comment]
> I might have had something to do with it. I filed the feature request back in 2010

Haha, so I see. :) So, the question is still there then. Should we treat Views special in the first place?

> I've found a whole lot wrong with the order of the entries in the admin menu and am currently creating a meta-issue

Cool, that'd be great.

**Comment by klonos** [comment]
> Should we treat Views special in the first place?

We do have ```Structure``` -> ```Content types``` -> ```[content_type]``` -> ```[operations]``` so we are not special-casing. The feature request was to bring views in parity with content types and provide easy(ier) access to each view action/operation.

**Comment by quicksketch** [comment]
Hm, yeah let me take that back. We're currently special casing the following:

- Text formats
- Menus
- Node types
- Listing fields within entity types (node types, vocabularies)
- Vocabularies
- Views

By "special casing" I mean we're augmenting the "normal" menu (as provided by hook_menu() and the Administration menu configuration) with dynamic entries. All of these implementations are in admin_bar.map.inc.

**Comment by klonos** [comment]
Let's keep this discussion for the specific menu and its respective listing page and talk in general about the menu entries in the meta...

**Comment by klonos** [comment]
I think that as a first step we should move all views entries under ```Structure``` -> ```Views``` -> ```List views``` and make sure that they are listed alphabetically.

**Comment by klonos** [comment]
...next we need to implement #525 (plain ```List``` becomes ```List views```) and #399 (```Add new view``` becomes ```Add view```).

**Comment by klonos** [comment]
...finally, I want to propose two things:

1. If you click on the ```Structure``` -> ```Views``` -> ```Settings```, you are taken to the basic settings page/tab. So how about we simplify the structure by removing the link to the basic settings and leave only the advanced settings as a submenu?

2. Similarly, clicking on a view name, you are already take to the view edit mode (besides, there is no "view" mode for views other than the list). So I see no reason for the ```Edit view``` submenu.

**Comment by klonos** [comment]
If you combine all the above, we move from this:

-Structure
  ...
  - Views
    - Add new view
    - Import
    - List
    - Admin Content (Content)
      - Edit view
    - Settings
      - Basic
      - Advanced
    - Taxonomy term (Content)
      - Edit view
    - Recent comments (Comment)
      - Edit view
    - Administer users (User)
      - Edit view
    - Front page (Content)
      - Edit view

**Comment by klonos** [comment]
...to this:

- Structure
  - ...
  - Views
    - Add view
    - Import
    - List views
      - Edit 'Front page' view (Content)
      - Edit 'Manage content' view (Content)
      - Edit 'Manage user accounts' view (User)
      - Edit 'Recent comments' view (Comment)
      - Edit 'Taxonomy term' view (Content)
      - Edit '[view_name]' view ([view_type])
      - ...
    - Settings
      - Advanced

**Comment by jenlampton** [comment]
:+1:

**Comment by mikemccaffrey** [comment]
+1 for moving the individual views under the "List views" menu item. 

-1 for removing the Basic item under settings. I feel that whenever we have a submenu, we should include the destination of the parent item as the first option for clarity's sake. Perhaps this is a broader discussion that needs to take place to define the proper conventions.

**Comment by jenlampton** [comment]
I moved views under "list views" in the admin bar, and also sorted the views alphabetically on the views UI so that they match the admin bar. PR at https://github.com/backdrop/backdrop/pull/1163

**Comment by quicksketch** [comment]
Here's a screenshot of the current PR:

![selection_030](https://cloud.githubusercontent.com/assets/100206/12095173/d2a26416-b2c1-11e5-9244-05b6026f8277.png)

Some things:

- We probably don't need "Edit View" under every view in the Admin bar.
- For some reason the order between the Views page and the Admin Bar don't match still.
- I'm not sure about the new markup around tags:
```
 +      'data' => check_plain($view->description) . '<div>' . t('<strong>Tag:</strong> ') . check_plain($view->tag) . '</div>', 
```
An unclassed div and strong tag? Here's a screenshot of the new mobile listing, the emphasis on the word Tags seems out of place.
![selection_031](https://cloud.githubusercontent.com/assets/100206/12095299/8fe1d5f2-b2c2-11e5-8239-347406316ab5.png)

**Comment by docwilmot** [comment]
Could also move the tag to the "View name" column (which incidentally is mislabelled too as there's a bunch more stuff than the name in there):
```
Display: Page
User-created
Type: node
Tag: default
```

Aside: what is "Tag" anyway? Can't find how to affect that in the UI anywhere.

**Comment by jenlampton** [comment]
> what is "Tag" anyway? Can't find how to affect that in the UI anywhere.

Tags are useful when you have a bunch of views and need to be able to tell them apart. I use them all the time in custom modules so I can see where they come from.  Use case: all commerce views could be tagged 'commerce' and then I know which ones I created vs which ones came with the commerce module. 

You can change tags under the "view name /description" area in the UI.

**Comment by jenlampton** [comment]
@quicksketch what is good markup for the tags? Class the div and remove the strong-tag?

**Comment by quicksketch** [comment]
> @quicksketch what is good markup for the tags? Class the div and remove the strong-tag?

Sure, though if we're putting the description next to it, that should probably be a div as well, and the label should be individually wrapped.

<code>
'data' => '<div class="views-description">' . check_plain($view->description) . '</div><div class="views-tag"><span class="views-tag-label">' . t('Tag:') . '</span>' . check_plain($view->tag) . '</div>',
</code>

Though overall the end result is that nothing is styled by default.

**Comment by jenlampton** [comment]
Pushed a new change that uses the recommended markup changes.

**Comment by quicksketch** [comment]
I manually merged this into 1.x at https://github.com/backdrop/backdrop/commit/b8aea75fc5141179d25e19d2a8dca137e0df4750.

But I excluded the "Move views under List" portion of the patch. If we want to do that, we'll need to consider if we should do it elsewhere. Right now all of our extra lists (Views, Vocabularies, Layouts, Menus, etc) are all in the top-list of the admin bar, not under their respective "List x" items. If we want to change that, let's make a separate issue and do it all at once.
